### ISSUE TITLE
Only first parameter handler is executed for a given parameter

### ISSUE THREAD

**Issue by alexwishart** [IBM-Swift/Kitura#1209]
If you register more than one handler to a parameter, e.g. 
```swift 
router.parameter(["id"], handlers: [handleInt, handleItem]) 
```
only the first handler is run.

**Comment by alexwishart** [comment]
Looks like parameters are being incorrectly removed in `RouterParameterWalker.swift` so when the `handle()` function is called again (which should allow for additional handlers to be called) the list of parameters is empty and `callback()` is executed prematurely.

**Comment by ianpartridge** [comment]
Can this be closed now? Looks like it was fixed in #1210

**Comment by alexwishart** [comment]
Yep, closing now 🙂
